### PR TITLE
Cleanup tx queue interface and address warnings/false positives from static analysis

### PR DESCRIPTION
**Pull request by marta-lokhova**

### CODE REVIEW

File: src/herder/HerderImpl.cpp [hunk 1]
```diff
@@ -2132,14 +2120,6 @@ HerderImpl::start()
     }
 
     restoreUpgrades();
-    // make sure that the transaction queue is setup against
-    // the lcl that we have right now
-    mTransactionQueue.maybeVersionUpgraded();
```
Comment (SirTyson) [inline src/herder/HerderImpl.cpp:2137]: Why was this removed? I assume the current ledger version is no longer required after all the refactoring we've been doing?

File: src/overlay/test/FloodTests.cpp [hunk 2]
```diff
@@ -167,12 +167,15 @@ TEST_CASE("Flooding", "[flood][overlay][acceptance]")
         auto ackedTransactions = [&](std::shared_ptr<Application> app) {
             // checks if an app received all transactions or not
             size_t okCount = 0;
+            auto& herder = static_cast<HerderImpl&>(app->getHerder());
+
             for (auto const& s : sources)
             {
-                okCount +=
```
Comment (SirTyson) [inline src/overlay/test/FloodTests.cpp:172]: I'm not a huge fan of this change, as it looks like we've made `getMaxSeqInPendingTxs` more involved. I see you have to use a `HerderImpl` cast to hide it under the `BUILD_TESTS` flag. Personally, given  that `getMaxSeqInPendingTxs` is a const function (or rather a function that could be made const) I don't think it's super important to hide behind BUILD_TESTS. That being said, you're more familiar with the semantics of this function so I'll defer to you.

**Review by SirTyson — APPROVED** [review]
LGTM, a couple questions.

File: src/herder/HerderImpl.cpp (hunk 1 above)
Comment (marta-lokhova) [reply src/herder/HerderImpl.cpp:2137]: This is dead code at this point, as it's dealing with v13 protocol upgrade (this will never happen on a production network)

File: src/overlay/test/FloodTests.cpp (hunk 2 above)
Comment (marta-lokhova) [reply src/overlay/test/FloodTests.cpp:172]: Mmm, not sure what you mean here -`getMaxSeqInPendingTxs` was removed from Herder's API completely as it was used only in this one place, so I don't see much value in exposing it publicly.

Comment (SirTyson) [reply src/overlay/test/FloodTests.cpp:172]: I'm just not a huge fan of casting to an `impl` class, but it's once place in a test so I don't think it matters.

Comment (marta-lokhova) [reply src/overlay/test/FloodTests.cpp:172]: yeah it's not great, but it's really common across our tests. Ideally, we could have a wrapper or a subclass that properly exposes test-only functions to keep the actual herder API tidy, but that's more involved...

**Comment by dmkozh** [comment]
r+ 5626aba
